tests/sentry/event_manager/test_event_manager.py

```diff
@@ -2117,39 +2117,51 @@ def test_throws_when_matches_discarded_hash(self) -> None:
         )
         GroupHash.objects.filter(group=group).update(group=None, group_tombstone_id=tombstone.id)
 
-        manager = EventManager(
-            make_event(message="foo", event_id="b" * 32, fingerprint=["a" * 32]),
-            project=self.project,
-        )
-        manager.normalize()
+        from sentry.utils.outcomes import track_outcome
 
         a1 = CachedAttachment(name="a1", data=b"hello")
         a2 = CachedAttachment(name="a2", data=b"world")
 
-        cache_key = cache_key_for_event(manager.get_data())
-        attachment_cache.set(cache_key, attachments=[a1, a2])
-
-        from sentry.utils.outcomes import track_outcome
-
-        mock_track_outcome = mock.Mock(wraps=track_outcome)
-        with mock.patch("sentry.event_manager.track_outcome", mock_track_outcome):
-            with self.feature("organizations:event-attachments"):
-                with self.tasks():
-                    with pytest.raises(HashDiscarded):
-                        manager.save(self.project.id, cache_key=cache_key, has_attachments=True)
-
-        assert mock_track_outcome.call_count == 3
-
-        for o in mock_track_outcome.mock_calls:
-            assert o.kwargs["outcome"] == Outcome.FILTERED
-            assert o.kwargs["reason"] == FilterStatKeys.DISCARDED_HASH
-
-        o = mock_track_outcome.mock_calls[0]
-        assert o.kwargs["category"] == DataCategory.ERROR
-
-        for o in mock_track_outcome.mock_calls[1:]:
-            assert o.kwargs["category"] == DataCategory.ATTACHMENT
-            assert o.kwargs["quantity"] == 5
+        for i, event_id in enumerate(["b" * 32, "c" * 32]):
+            manager = EventManager(
+                make_event(message="foo", event_id=event_id, fingerprint=["a" * 32]),
+                project=self.project,
+            )
+            manager.normalize()
+            discarded_event = Event(
+                project_id=self.project.id, event_id=event_id, data=manager.get_data()
+            )
+
+            cache_key = cache_key_for_event(manager.get_data())
+            attachment_cache.set(cache_key, attachments=[a1, a2])
+
+            mock_track_outcome = mock.Mock(wraps=track_outcome)
+            with mock.patch("sentry.event_manager.track_outcome", mock_track_outcome):
+                with self.feature("organizations:event-attachments"):
+                    with self.tasks():
+                        with self.feature("organizations:grouptombstones-hit-counter"):
+                            with pytest.raises(HashDiscarded):
```

Comment:
You could combine all of these

```python
with (
  mock.patch(...),
  self.feature(...)
):
``` 

which can help with the indentation levels.